### PR TITLE
Do fire DNS tracing events

### PR DESCRIPTION
**Pull request by asvetlov**
Fix for #2841

### CODE REVIEW

File: tests/test_web_functional.py [hunk 1]
```diff
@@ -1663,20 +1668,50 @@ def make_handler(appname):
...
+                return await self._resolver.resolve(host, port, family)
 
-    client = await aiohttp_client(app, trace_configs=[trace_config])
+    resolver = FakeResolver({'example.com': server.port})
+    connector = aiohttp.TCPConnector(resolver=resolver)
+    client = aiohttp.ClientSession(connector=connector,
+                                   trace_configs=[trace_config])
 
-    await client.get('/redirector', data="foo")
+    await client.get('http://example.com/redirector', data="foo")
```
Comment (pfreixes) [inline tests/test_web_functional.py:1705]: a `f"localhost:{server.port}/redirector"` does not fire a DNS event? If it does we wouldn't need it the `FakeResolver`

Comment (asvetlov) [reply tests/test_web_functional.py:1705]: Resolver doesn't perform DNS lookup for IP addresses, only for domain names.
`localhost` should be sufficient but unfortunately there are too many boxes with declared but actually disabled IPv6 support.

Comment (pfreixes) [reply tests/test_web_functional.py:1705]: sorry I don't get the following sentence.

> there are too many boxes with declared but actually disabled IPv6 support.

Comment (pfreixes) [reply tests/test_web_functional.py:1705]: I dont want to block the PR just for a test that in any case can be at any moment refactorized.

Comment (asvetlov) [reply tests/test_web_functional.py:1705]: Ok, I'll try to describe. Some broken environments (Travis-CI included IIRC) resolves `localhost` to both `127.0.0.1` and `::1` but IPv6 stack is disabled.
By this communication over IPv6 `::1` fails.
That's why `aiohttp.test_utils` and aiohttp test suite use IPv4 `127.0.0.1` explicitly.
The situation could be improved I hope but working on it is definitely out if the issue scope.

**Pull request merged**